### ISSUE TITLE
Allow okhttp client to have a customer resolver

### ISSUE THREAD

**Issue by sbaum1994** [lightstep/lightstep-tracer-java-common#94]
issue opened for @dmichel1,

Similar thing was enabled for grpc (relevant commit is https://github.com/lightstep/lightstep-tracer-java-common/commit/17a2460cef9dba6f2547b8191820d9d4f7d421b3) 

This is a request to allow for a custom builder capability, in order to enable custom resolver behavior.

This is where we currently build the OkHttpClient:
https://github.com/lightstep/lightstep-tracer-java-common/blob/master/okhttp/src/main/java/com/lightstep/tracer/shared/HttpCollectorClient.java#L96

This is a code snippet of the customization necessary (code by @dmichel1 ):
```
private static OkHttpClient start(long deadlineMillis) {
    return new OkHttpClient.Builder()
      .connectTimeout(deadlineMillis, TimeUnit.MILLISECONDS)
      .dns(new RoundRobin())
      .build();
  }

  static private class RoundRobin implements Dns {
    @Override
    public List<InetAddress> lookup(String hostname) throws UnknownHostException {
      if (hostname == null) {
        throw new UnknownHostException("hostname == null");
      } else {
        try {
          final List<InetAddress> addresses = Arrays.asList(InetAddress.getAllByName(hostname));
          Collections.shuffle(addresses);
          return addresses;
        } catch (NullPointerException var4) {
          UnknownHostException unknownHostException =
            new UnknownHostException("Broken system behaviour for dns lookup of " + hostname);
          unknownHostException.initCause(var4);
          throw unknownHostException;
        }
      }
    }
  }
```
Basically, we'd need to enable in Options(?) the ability to pass in a custom dns implementation that is then set when the okhttp client is being built, or have the default okhttp resolver if not configured.

Default current code:
```
    private static OkHttpClient start(long deadlineMillis) {
        return new OkHttpClient.Builder()
                .connectTimeout(deadlineMillis, TimeUnit.MILLISECONDS)
                .build();
    }
```

Behavior `withCustomDns` passed in as an option: 
```
private static OkHttpClient startWithCustomDNS(long deadlineMillis, customDns Dns) {
    return new OkHttpClient.Builder()
      .connectTimeout(deadlineMillis, TimeUnit.MILLISECONDS)
      .dns(customDns)
      .build();
  }
```
